### PR TITLE
PAL: Fix missing context-restore in signal.cpp on FreeBSD

### PR DESCRIPTION
**Pull request by josteink**
This fixes build errors encountered on FreeBSD.

### CODE REVIEW

File: src/pal/src/exception/signal.cpp [hunk 1]
```diff
@@ -562,19 +562,22 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
...
-#elif __LINUX__
+#elif __LINUX__ || defined(__FreeBSD__)
                 // Here we shamelessly exploit undocumented behavior of Linux signal handlers:
                 // 
                 // When signal handler exits it completely restores context saved in ucontext, 
                 // if any changes are applied to it by the signal handler they will also be 
                 // restored and affect further course of execution.
                 //
+		// This also seems to hold true for FreeBSD, except FreeBSD is nice enough to
+		// document it: http://www.unix.com/man-page/FreeBSD/2/sigaction/
```
Comment (akoeplinger) [inline src/pal/src/exception/signal.cpp]: this uses tabs instead of spaces like the rest of the comment block

**Comment by josteink** [comment]
Good catch. Fixed now.

File: src/pal/src/exception/signal.cpp [hunk 2]
```diff
@@ -562,19 +562,22 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
...
+                // This also seems to hold true for FreeBSD, except FreeBSD is nice enough to
+                // document it: http://www.unix.com/man-page/FreeBSD/2/sigaction/
+                //
                 // We'd love to avoid using this assumption, but unfortunately setcontext()
                 // (which is used in PAL implementation of RtlRestoreContext) doesn't restore eflags 
                 // and loads zero into RAX register. It might be fine for exception handling, but unacceptable 
                 // for debugger. That's why for debugger's signals (breakpoint and singlestep) we
                 // actually allow HardwareExceptionHandler to return and change context on exit.
-
+
```
Comment (Eyas) [inline src/pal/src/exception/signal.cpp]: nit: Still tabbing issues here.

**Comment by josteink** [comment]
Right. That should be the end of subtle spacing-issues.

**Comment by kangaroo** [comment]
LGTM

**Comment by sergiy-k** [comment]
LGTM

**Comment by sergiy-k** [comment]
@josteink Could you please resolve conflicts in your change? Thank you!

**Comment by josteink** [comment]
After the [this commit](https://github.com/dotnet/coreclr/commit/7de896c515ab61bf319e3372acd4f70ae354023f) was merged, signal.cpp is no longer a build-breaker and it seems this PR is no longer required.

Closing for now.

**Pull request closed without merging**